app/models/assignment_properties.rb

```diff
@@ -126,4 +126,18 @@ def not_timed_and_scanned
     msg = I18n.t('activerecord.errors.models.assignment_properties.attributes.is_timed.not_scanned')
     errors.add(:base, msg) if is_timed && scanned_exam
   end
+
+  # Ensure remote_autotest_settings_id is unique for a given autotester
+  def remote_autotest_settings_id_check
+    unless self.remote_autotest_settings_id.nil?
+      assign_prop = AssignmentProperties.joins(assignment: :course).where(
+        remote_autotest_settings_id: self.remote_autotest_settings_id,
+        course: { autotest_setting_id: self.assignment.course.autotest_setting_id }
+      )
+      if assign_prop.count == 1 && self.id != assign_prop.first.id
+        errors.add(:remote_autotest_settings_id,
+                   'remote_autotest_settings_id has already been taken by the same autotester.')
```

Comment:
Make this an internationalized string. You can add it to the corresponding `models` YML file.